Remove the need for bundlers in the build pipeline

davxdo opened the pull request:
- build(rollup): remove rollup and related files
- build(tsc): create cjs and esm bundles
- chore(package.json): remove browser and browserlist

As per the last commit, I perceive we'll have issues with the @solana/web3.js lib that depends on some node builtins.
An error like `process is not defined` or `buffer is not defined` is expected. 

Unsure on how best to handle that, we could list those as peer deps, or for the sake of simplicity, bundle the @solana/web3.js package to have those builtins provided, so upstream users don't have to be concerned about it.

Similar to the rollup config that was part of this lib

davxdo committed:
build(rollup): remove rollup and related files

Step one of removing a bundler from the build system

davxdo committed:
build(tsc): create cjs and esm bundles

davxdo committed:
chore(package.json): remove browser and browserlist

I dont think these have a role to play now, but i percieve we may have problems with downstream
packages (@solana/web3.js) that depend on node builtins, not sure how best to handle that yet. Maybe
listed in peer deps?

jordaaash commented on tsconfig.json:
```diff
@@ -13,7 +13,8 @@
         "isolatedModules": true,
-        "typeRoots": ["node_modules/@types", "types"]
+        "typeRoots": ["node_modules/@types", "types"],
+        "skipLibCheck": true
```
Do we need `skipLibCheck`?

davxdo replied:
Possibly not, made a small mistake here with using `npm` instead of `yarn`, so some transient deps may be using the wrong versions. I was getting errors running a build from incorrect types in a `@types/...` package. Will switch to yarn and see if it's still needed.

jordaaash reviewed (approved):
Looks good!

davxdo committed:
fix(package manager): remove package.lock

incorrect use of npm instead of yarn

davxdo committed:
build(cjs/esm): add postbuild script

delegate the module type check to package.json within lib/esm and lib/cjs

davxdo commented on package.json:
```diff
@@ -16,24 +16,17 @@
         "clean": "shx rm -rf lib",
-        "build": "yarn clean && tsc && rollup -c",
+        "build": "yarn clean && tsc -p tsconfig.json && tsc -p tsconfig-cjs.json",
+        "postbuild": "echo '{\"type\":\"commonjs\"}' > lib/cjs/package.json && echo '{\"type\":\"module\"}' > lib/esm/package.json",
```
this was the lesser of the two evils. Cleaner than changing all the extensions in the `lib/cjs` directory

davxdo committed:
build(package.json): retain type:module

mainly for running the test script, otherwise it will load the src directory files as commonJS

davxdo commented:
This is good to go. Tested both esm/cjs files with success
